src/log.rs

```diff
@@ -43,10 +43,16 @@ impl Log {
         let path_str = path;
         let path = PathBuf::from(path);
 
-        if !path.exists() && !create || path.exists() && !path.is_dir() {
-            return Err(Error::InvalidPath(path_str.to_string()));
+        if create {
+            if path.exists() && !path.is_dir() {
+                return Err(Error::InvalidPath(path_str.to_string()));
+            } else if !path.exists() {
+                fs::create_dir(&path)?;
+            }
         } else {
-            fs::create_dir(&path)?;
+            if !path.exists() {
```

Comment:
I think should be instead `if !path.is_dir()` which returns `false` either if the path doesn't exist or if its not pointing to a directory.